source/common/http/alternate_protocols_cache_manager_impl.cc

```diff
@@ -0,0 +1,39 @@
+#include "common/http/alternate_protocols_cache_manager_impl.h"
+
+#include "common/http/alternate_protocols_cache_impl.h"
+#include "common/protobuf/protobuf.h"
+
+#include "absl/container/flat_hash_map.h"
+
+namespace Envoy {
+namespace Http {
+
+SINGLETON_MANAGER_REGISTRATION(alternate_protocols_cache_manager);
+
+AlternateProtocolsCacheSharedPtr AlternateProtocolsCacheManagerImpl::getCache(
+    const envoy::config::core::v3::AlternateProtocolsCacheOptions& options) {
+  const auto& existing_cache = caches_.find(options.name());
+  if (existing_cache != caches_.end()) {
+    if (!Protobuf::util::MessageDifferencer::Equivalent(options, existing_cache->second.options_)) {
+      throw EnvoyException(
+          fmt::format("options specified alternate protocols cache '{}' with different settings",
```

Comment:
worth logging both sets of options since we have them?

Reply:
Done!